plugins/out_s3/s3.c

```diff
@@ -2117,17 +2132,35 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
 
     /* Process chunk */
     if (ctx->log_key) {
-        chunk = flb_pack_msgpack_extract_log_key(ctx,
-                                                 event_chunk->data,
-                                                 event_chunk->size);
+        chunk_tmp = flb_pack_msgpack_extract_log_key(ctx,
+                                                     event_chunk->data,
+                                                     event_chunk->size);
     }
     else {
-        chunk = flb_pack_msgpack_to_json_format(event_chunk->data,
-                                                event_chunk->size,
+        chunk_tmp = flb_sds_create_len(event_chunk->data, event_chunk->size);
+
+    }
+
+    if (ctx->format == S3_FORMAT_JSON) {
+        chunk = flb_pack_msgpack_to_json_format(chunk_tmp,
+                                                flb_sds_len(chunk_tmp),
                                                 FLB_PACK_JSON_FORMAT_LINES,
                                                 ctx->json_date_format,
                                                 ctx->date_key);
+        flb_sds_destroy(chunk_tmp);
+    }
+    else if (ctx->format == S3_FORMAT_CSV) {
+        /*
+         * CSV is a bit tricky, if the user asked to include the column names
+         * we need to know if the content will be placed in a new object or not,
+         * for hence the formatting of the CSV will happen inside the function
+         * s3_store_buffer_put().
+         *
+         * For now, we duplicate the buffer for safety reasons.
```

Comment:
what safety does it bring? I am not understanding why we need to duplicate it here...

Reply:
I think this comment is also outdated, with respect to the refactoring done to `s3_format`.